README.md

```diff
@@ -2,10 +2,31 @@
 
 # State management in iOS
 
+## Introduction
+
+In iOS development it's always good to choose a right architecture.
+As you know in projects used `MVC` pattern it becomes hard
+to maintain and test massive view controller after some time.
+
+Let's try to make it more clean and reusable using one way data flow architecture.
+
+### Unidirectional Data Flow pattern
+
+Main aspects:
+
+* The whole application `state` is stored in single object (store)
+* The only way to change the `state` is to trigger (dispatch) an `action` (an object describing what to do)
+* `Reducer` specifies how to generate new `state` based on given `action` and current `state`.
+
+That's basically it.
+
+In this tutorial you will see how to build sample iOS application
+using exiting unidirectional framework.
+
 ## About application
 
-We are going to build universal application to allow users track they frequently data.
-It could be let's say mileage each month or outcome each day.
+We are going to build simple application to allow users track they frequently data.
```

Comment:
I would say "We are going to build simple application that allows users track their frequently collected data."